examples/platforms/cc2538/random.c

```diff
@@ -28,60 +28,107 @@
 
 /**
  * @file
- *   This file implements a pseudo-random number generator.
+ *   This file implements a random number generator.
  *
- * @warning
- *   This implementation is not a true random number generator and does @em satisfy the Thread requirements.
  */
 
 #include <openthread-types.h>
 
 #include <common/code_utils.hpp>
+#include <platform/radio.h>
 #include <platform/random.h>
 #include "platform-cc2538.h"
 
-static uint32_t s_state = 1;
+static void generateRandom(uint16_t aInputLength, uint8_t *aOutput, uint16_t *aOutputLength)
+{
+    uint16_t seed = 0;
+
+    HWREG(SOC_ADC_ADCCON1) &= ~(SOC_ADC_ADCCON1_RCTRL1 | SOC_ADC_ADCCON1_RCTRL0);
+    HWREG(SYS_CTRL_RCGCRFC) = SYS_CTRL_RCGCRFC_RFC0;
+
+    while (HWREG(SYS_CTRL_RCGCRFC) != SYS_CTRL_RCGCRFC_RFC0);
+
+    HWREG(RFCORE_XREG_FRMCTRL0) = RFCORE_XREG_FRMCTRL0_INFINITY_RX;
+    HWREG(RFCORE_SFR_RFST) = RFCORE_SFR_RFST_INSTR_RXON;
+
+    while (!HWREG(RFCORE_XREG_RSSISTAT) & RFCORE_XREG_RSSISTAT_RSSI_VALID);
+
+    for (uint16_t index = 0; index < aInputLength; index++)
+    {
+        if (!(index % 2))
```

Comment:
You can simplify this further by removing this if check and just iterating over the individual bytes.